File: problem2/problem2.c
#include <stdio.h>
/*
Taken from problem 2
find the sum of all even fibonacci numbers
less than 4 million
*/
int main(){
    //p1 and p2 hold the previous two values in the fibonacci sequence
    //current holds the lastest value in the fibonacci sequence
    int p1 = 0;
    int p2 = 1;
    int current = 0;
    int sum = 0;
    //continue looping until it is greater than 4 million
    while (current < 4000000)
    {
        //if it is even add it to sum
        if(current%2 == 0){
            sum+=current;
        }
        //add in previous two values to get new value
        current = p1+p2;
        p1 = p2;
        p2 = current;
    }   
    printf("The sum of all even fibonacci numbers is %d",sum);
    return 0;
}

File: problem2/problem2.py
#Michael Vogt
#Taken from problem 2
#find the sum of all even fibonacci numbers
#less than 4 million
def even_fibonacci():
    #set up array for dynamic fibonacci
    fib = []
    fib.append(0)
    fib.append(1)
    #used to determine location in fib array
    counter = 2
    sum = 0
    flag = True
    #countinue looping untill the last value in fibonacci is greater than 4 million
    while(flag):
        #add the previous two values together
        fib.append(fib[counter-1]+fib[counter-2])
        if fib[counter] > 4000000:
            flag = False
        #determine if it is even and add it to the sum if it is
        elif fib[counter]%2 == 0:
            sum = sum+fib[counter]
        counter = counter+1
    return sum
print("The sum of all even fibonacci numbers is",end=' ')
print(even_fibonacci())